### PR TITLE
fix(cli): add aliases to monorepo paths to tsconfig includes

### LINKED ISSUE #21996: Importing components with alias in monorepo causes VSCode/Volar to fail finding modules/type declarations
### Environment

- Operating System: Darwin
- Node Version:     v18.16.0
- Nuxt Version:     3.6.1
- Nitro Version:    2.5.2
- Package Manager:  pnpm@8.6.6
- Builder:          vite
- User Config:      -
- Runtime Modules:  -
- Build Modules:    -

### Reproduction

https://stackblitz.com/github/samydoesit/nuxt-monorepo-import-reproduction
https://github.com/samydoesit/nuxt-monorepo-import-reproduction

### Describe the bug

Hi, i am experiencing issues while importing components in a monorepo using an alias.

The problem arises when using VSCode with the Volar extension. Since our ui directory is located outside of the Nuxt project directory, VSode (Volar) is unable to locate the module or type declaration.

However, if i manually add the path of the alias to the "includes" field in nuxt's tsconfig, the resolution in VSCode works.

### Additional context

_No response_

### Logs

```shell-script
Cannot find module '@ui/components/button/Button.vue' or its corresponding type declarations.
```

### PR DESCRIPTION
**Pull request by samydoesit**
### 🔗 Linked issue

resolves #21996

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Add aliases to include property of tsconfig so that VSCode ts and intellisense works fine in a monorepo.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxi/src/utils/prepare.ts [hunk 1]
```diff
@@ -68,6 +69,10 @@ export const writeTypes = async (nuxt: Nuxt) => {
     }
   }
 
+  for (const alias in nuxt.options.alias) {
+    tsConfig.include.push(aliases[alias])
+  }
```
Comment (danielroe) [inline packages/nuxi/src/utils/prepare.ts:74]: can we deduplicate this somehow? For example, we shouldn't add any aliases that are already covered by an existing glob.

File: packages/nuxi/src/utils/prepare.ts [hunk 2]
```diff
@@ -68,6 +69,13 @@ export const writeTypes = async (nuxt: Nuxt) => {
     }
   }
 
+  for (const alias in nuxt.options.alias) {
+    if (aliases[alias].startsWith(nuxt.options.srcDir)) {
+      continue
+    }
+    tsConfig.include.push(aliases[alias])
+  }
```
Comment (danielroe) [inline packages/nuxi/src/utils/prepare.ts:77]: let's move this up into the previous for/in loop so we don't iterate twice over the aliases. We might also want to consider having different handling for directories (maybe need glob)?

Comment (samydoesit) [reply packages/nuxi/src/utils/prepare.ts]: moved it up and aligned with previous alias handling in the for loop.
If it's a directory is it necessary to also add the path with `/*`?

**Review by danielroe — APPROVED** [review]

**Comment by danielroe** [comment]
oops, my mistake - didn't mean to merge this with co-authored-by 🤦

**Pull request merged**